src/Plugin/resource/ResourceEntity.php

```diff
@@ -183,9 +186,21 @@ protected function processPublicFields(array $field_definitions) {
     // The fields that only contain a property need to be set to be
     // ResourceFieldEntity. Otherwise they will be considered regular
     // ResourceField.
-    return array_map(function ($field_definition) {
+    $field_definitions = array_map(function ($field_definition) {
       return $field_definition + array('class' => '\Drupal\restful\Plugin\resource\Field\ResourceFieldEntity');
     }, $field_definitions);
+
+    // If there is an alternate id field, use it instead of the entity id.
+    $plugin_definition = $this->getPluginDefinition();
```

Comment:
i'd be happy to remove this part, and keep all declarations in publicFields() without auto-removing the ID. maybe one would like to expose the uuid, but keep the id in place.